proposals/text/5289-recency-reranker.md

```diff
@@ -0,0 +1,57 @@
+- Title: Addition of a 'RecencyReranker' node
+- Decision driver: @elundaeva
+- Start Date: 2023-07-05
+- Proposal PR: https://github.com/deepset-ai/haystack/pull/5289
+- Github Issue or Discussion: some links available in the "Detailed design" section below
+
+# Summary
+
+This reranker allows to have retrieved documents sorted not only by relevance (default) but also with recency factored in.
+
+# Basic example
+
+The RecentnessReranker would be instantiated as follows:
+
+  ``` python
+
+	  reranker = RecentnessReranker(
+	      date_identifier="date",
+	      weight="0.5",
+          top_k=3,
+	      method="reciprocal_rank_fusion",
+	  )
+  ```
+
+# Motivation
+
+Initially this reranker was implemented by Timo for a customer case where the date of the document mattered for retrieval. The reason we would like to add it to Haystack is because we see wider use for this node in future customer cases. One example where document recency matters is in a QA solution based on technical documentation with release notes of a software product - the older release notes should naturally have less priority in the responses than the most recent ones. 
+
+# Detailed design
+
+The reranker has already been implemented by Timo here: https://github.com/deepset-ai/deepset-cloud-custom-nodes/blob/main/deepset_cloud_custom_nodes/rankers/recentness_reranker.py & PR was reviewed at the time by Seb and Florian: https://github.com/deepset-ai/deepset-cloud-custom-nodes/pull/54. 
+
+Additionally, you can see the code for this proposal here: https://github.com/deepset-ai/haystack/pull/5301/files. It is the same code as above, just with small naming changes (e.g. "rff" method got changed to "reciprocal_rank_fusion" to match the existing naming used Haystack's JoinDocuments node)
+
+As a general description, the reranker works by sorting the documents based on date and modifying the relevance score calculation slightly to include recency-based weight.
+
+# Drawbacks
+
+Since this is a relatively small change without any effect on existing nodes, I do not see major reasons not to add this reranker. The only important limitation to using this node is the need to have a metadata field with document date already present and for the "score" method the need to double-check that your retreiver node outputs a score within [0,1] range. 
+
+# Alternatives
+
+Without adding this feature it will not be possible to handle customer cases where recency of documents matters for the response, see examples in the Motivation section.
```

Comment:
```suggestion
Without adding this feature it will not be possible to handle customer and community cases where recency of documents matters for the response, see examples in the Motivation section.
```